lib/collection/src/collection_manager/holders/proxy_segment.rs

```diff
@@ -381,8 +383,31 @@ impl SegmentEntry for ProxySegment {
         true
     }
 
-    fn flush(&self) -> OperationResult<u64> {
-        Ok(self.wrapped_segment.get().read().version())
+    fn flush(&self) -> OperationResult<SeqNumberType> {
+        let deleted_points_guard = self.deleted_points.read();
+        let deleted_indexes_guard = self.deleted_indexes.read();
+        let created_indexes_guard = self.created_indexes.read();
+
+        if deleted_points_guard.is_empty()
+            && deleted_indexes_guard.is_empty()
+            && created_indexes_guard.is_empty()
```

Comment:
Why not?
```suggestion
        if self.deleted_points.read().is_empty()
            && self.deleted_indexes.read().is_empty()
            && self.created_indexes.read().is_empty()
```

Reply:
Need to keep guard to avoid race condition - otherwise `deleted_points` could change after condition check but before flush.